spark/src/main/java/org/apache/iceberg/spark/arrow/ArrowUtils.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.arrow;
+
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.types.DateUnit;
+import org.apache.arrow.vector.types.FloatingPointPrecision;
+import org.apache.arrow.vector.types.TimeUnit;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.spark.sql.types.ArrayType;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.DecimalType;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+public class ArrowUtils {
+
+  private static ArrowUtils instance;
+  private RootAllocator rootAllocator;
+
+  private ArrowUtils() {
+    rootAllocator = new RootAllocator(Long.MAX_VALUE);
+  }
+
+  public static ArrowUtils instance() {
+    if (instance == null) {
+      instance = new ArrowUtils();
+    }
+    return instance;
+  }
+
+  public RootAllocator rootAllocator() {
+    return rootAllocator;
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  public DataType fromArrowType(ArrowType data) {
```

Comment:
Why convert directly from Arrow to Spark? Shouldn't this make guarantees about Arrow storage fields that are used for a given Iceberg type instead?